aptos-move/move-examples/fungible_asset/sources/fungible_asset.move

```diff
@@ -0,0 +1,377 @@
+/// This defines the fungible asset module that can issue fungible assets of any `FungibleSource` object. The source
+/// can be a token object or any object that equipped with `FungibleSource` resource.
+module fungible_asset::fungible_asset {
+    use aptos_std::smart_table::{Self, SmartTable};
+    use aptos_framework::object::{Self, Object, object_address, generate_transfer_ref, DeleteRef, generate_delete_ref};
+    use std::error;
+    use std::signer::address_of;
+    use std::option;
+    use std::option::Option;
+    #[test_only]
+    use aptos_framework::object::ConstructorRef;
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+    #[test_only]
+    use std::signer;
+
+    friend fungible_asset::fungible_source;
+
+    /// The coin resource existence error.
+    const EFUNGIBLE_ASSET: u64 = 1;
+    /// Amount cannot be zero.
+    const EAMOUNT_CANNOT_BE_ZERO: u64 = 2;
+    /// Not the owner.
+    const ENOT_OWNER: u64 = 3;
+    /// The token account has asset mismatch.
+    const EASSET_ADDRESS_MISMATCH: u64 = 4;
+    /// The token account has positive amount so cannot be deleted.
+    const EBALANCE_NOT_ZERO: u64 = 5;
+    /// The token account is still frozen so cannot be deleted.
+    const ESTILL_FROZEN: u64 = 6;
+    /// The pinned fungible asset object existence error.
+    const PINNED_EFUNGIBLE_ASSET_OBJECT: u64 = 7;
+    /// Insufficient amount.
+    const EINSUFFICIENT_BALANCE: u64 = 8;
+    /// FungibleAsset type mismatch.
+    const EFUNGIBLE_ASSET_TYPE_MISMATCH: u64 = 9;
+
+
+    /// Represents all the fungible asset objects of an onwer keyed by the address of the base asset object.
+    struct FungibleAssetStore has key {
+        index: SmartTable<address, Object<PinnedFungibleAsset>>
+    }
```

Comment:
per convo, let's extract this and make it part of a (User|Account|Primary)FungibleAssets